### PR TITLE
Add sever url in agentcore

### PR DESCRIPTION
**Pull request by qiujian16**
Need this to be set from API

### CODE REVIEW

File: nucleus/v1/types.go [hunk 1]
```diff
@@ -97,6 +97,22 @@ type SpokeCoreSpec struct {
 	// The spoke agent generates a random name if it is not set, or discovers the appropriate cluster name on openshift.
 	// +optional
 	ClusterName string `json:"clusterName,omitempty" protobuf:"bytes,4,opt,name=clusterName"`
+
+	// ExternalAPIServer represents the apiserver url and ca bundle that is accessible externally
+	// +optional
+	ExternalAPIServer ServerURL `json:"externalSeverURL,omitempty" protobuf:"bytes,5,opt,name=externalSeverURL"`
```
Comment (deads2k) [inline nucleus/v1/types.go]: I'm thinking this should be a slice to accommodate multiples and you should explain what empty slice means.

Comment (pmorie) [reply nucleus/v1/types.go]: Typo: `externalServerURL`

Agree that this should be a slice.

Comment (qiujian16) [reply nucleus/v1/types.go]: update to slice, also change the `spokeClientConfig` on spoke cluster to slice.

Comment (qiujian16) [reply nucleus/v1/types.go]: update to slice, also change the `spokeClientConfig` on spoke cluster to slice.

**Pull request merged**